libnavui-shield/src/main/java/com/mapbox/navigation/ui/shield/RoadShieldContentManagerImpl.kt

```diff
@@ -0,0 +1,274 @@
+package com.mapbox.navigation.ui.shield
+
+import com.mapbox.api.directions.v5.models.BannerComponents
+import com.mapbox.bindgen.Expected
+import com.mapbox.bindgen.ExpectedFactory
+import com.mapbox.navigation.ui.shield.internal.model.RouteShieldToDownload
+import com.mapbox.navigation.ui.shield.model.RouteShieldError
+import com.mapbox.navigation.ui.shield.model.RouteShieldOrigin
+import com.mapbox.navigation.ui.shield.model.RouteShieldResult
+import com.mapbox.navigation.utils.internal.InternalJobControlFactory
+import kotlinx.coroutines.CancellationException
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.suspendCancellableCoroutine
+import java.util.UUID
+import kotlin.coroutines.resume
+
+/**
+ *
+ * The object makes use of the data from [BannerComponents.imageBaseUrl] and
+ * [BannerComponents.mapboxShield] to download legacy road shields or mapbox designed road shields
+ * if the values are available. The logic on how the utility decided which one to use is described
+ * below:
+ *
+ * For a given sample instruction
+ *
+ * bannerInstructions: {
+ *    primary: {
+ *      components: [
+ *          {
+ *              type: text
+ *              text: Fremont Street
+ *          },
+ *          {
+ *              imageBaseUrl: https://mapbox.navigation.shields.s3.....
+ *              mapbox_shield: {
+ *                  base_url: https://api.mapbox.com/styles/v1
+ *                  text_color: black
+ *                  display_ref: 880
+ *                  shieldName: us-interstate
+ *              }
+ *              type: icon
+ *              text: I880
+ *          }
+ *      ]
+ *    }
+ * }
+ *
+ * 1. If imageBaseUrl is non null, mapbox shield is null:
+ *     - Request shield using imageBaseUrl:
+ *       - If request success: return legacy shield
+ *       - If request fails: return error
+ * 2. If mapbox shield is non null, imageBaseUrl is null
+ *     - If sprite does not exist for shield url: return error
+ *       - If sprite exists:
+ *         - If placeholder does not exist: return error
+ *         - If placeholder exists:
+ *           - Request shield using mapbox shield:
+ *             - If request success: return mapbox shield
+ *             - If request fails: return error
+ * 3. If mapbox shield & imageBaseUrl is non null
+ *     - If sprite does not exist for shield url: return error
+ *       - If sprite exists:
+ *         - If placeholder does not exist: return error
+ *         - If placeholder exists:
+ *           - Request shield using mapbox shield:
+ *             - If request success: return mapbox shield
+ *             - If request fails: repeat step 1.
+ */
+internal class RoadShieldContentManagerImpl(
+    private val shieldResultCache: ShieldResultCache = ShieldResultCache()
+) : RoadShieldContentManager {
+    companion object {
+        internal const val CANCELED_MESSAGE = "canceled"
```

Comment:
```suggestion
    private companion object {
        private const val CANCELED_MESSAGE = "canceled"
```

Reply:
Used in test so cannot make it private
@Guardiola31337

Reply:
You can, I tested locally - unless my Gradle cache is bugging me 😅

Reply:
Yeah, Gradle / AS were trolling me. In any case we can add `internal` to `companion object`.